duckduckgo/duckduckgo-privacy-extension
# Split up the declarative-net-request module

Initially the declarative-net-request module was only used to manage
declarativeNetRequest rulesets generated for the extension
configs (the tds.json tracker block list and extension-config.json
configuration). As we ported more features to the
declarativeNetRequest API, the module grew in scope considerably and
has now become unwieldy.

Let's improve the situation by:
 - Renaming the declarative-net-request module to
   dnr-config-rulesets.
 - Splitting out generally shared code into a new dnr-utils module.
 - Split out other code into different modules as it makes sense.
 - Add a comment in dnr-utils to explain how the rule IDs work, since
   that has gotten quite confusing for new contributors.

**Reviewer:** @jdorweiler 
**CC:** @sammacbeth 


## Description:



## Steps to test this PR:
1. Ensure the unit and integration tests still pass.
2. Test the Chrome MV3 build of the extension still works correctly.

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
